pkg/clients/azure/resourcegroup.go

```diff
@@ -0,0 +1,51 @@
+// /*
+// Copyright 2018 The Crossplane Authors.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// */
+
+package azure
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
+)
+
+// CreateOrUpdateGroup either creates a resource group or updates an already existing one by the same name
+func CreateOrUpdateGroup(client *Client, name string, location string) error {
+	if err := CheckResourceGroupName(name); err != nil {
+		return err
+	}
+	groupsClient := resources.NewGroupsClient(client.SubscriptionID)
```

Comment:
when you implement `sync` and `delete`, you should probably factor out this groups client instantiation so it can be done once and the client can be reused for other calls to the Azure API.  An example of this would be in the GKE controller's `connect` function, where it will return a `NewClusterClient` that will then be used for the rest of that reconciliation loop:
https://github.com/crossplaneio/crossplane/blob/master/pkg/controller/gcp/compute/gke.go#L172